filebeat/harvester/reader/multiline.go

```diff
@@ -113,40 +115,40 @@ func (mlr *Multiline) Next() (Message, error) {
 }
 
 func (mlr *Multiline) readFirst() (Message, error) {
-	for {
-		message, err := mlr.reader.Next()
-		if err != nil {
-			// no lines buffered -> ignore timeout
-			if err == sigMultilineTimeout {
-				continue
-			}
+	//for {
+	message, err := mlr.reader.Next()
+	if err != nil {
+		// no lines buffered -> ignore timeout
+		//if err == sigMultilineTimeout {
+		//	continue
+		//}
 
-			// pass error to caller (next layer) for handling
-			return message, err
-		}
+		// pass error to caller (next layer) for handling
+		return message, err
+	}
 
-		if message.Bytes == 0 {
-			continue
-		}
+	//if message.Bytes == 0 {
+	//	continue
+	//}
 
-		// Start new multiline event
-		mlr.clearBuffer()
-		mlr.resetBuffer(message)
-		mlr.setState((*Multiline).readNext)
-		return mlr.readNext()
-	}
+	// Start new multiline event
+	mlr.clearBuffer()
+	mlr.resetBuffer(message)
```

Comment:
doesn't resetBuffer imply clearBuffer?

Why don't we clear the buffer when finalizing the message? 

Just minor details.